File: verification/verify_fixes.js

const { chromium } = require('playwright');
const path = require('path');

(async () => {
    const browser = await chromium.launch();
    const page = await browser.newPage();
    await page.setViewportSize({ width: 1280, height: 960 });

    await page.goto('http://localhost:3000');
    await page.waitForTimeout(1000);

    // Test Dark Mode
    await page.click('#theme-toggle-btn');
    await page.waitForTimeout(500);

    // Check input backgrounds in Dark Mode (specifically focus)
    await page.fill('#app-name', 'Test App');
    await page.focus('#app-name');
    await page.screenshot({ path: 'verification/theme_check_v2/dark_mode_focus.png' });

    // Check Header
    await page.screenshot({ path: 'verification/theme_check_v2/dark_mode_header.png', clip: { x: 0, y: 0, width: 1280, height: 400 } });

    // Test New Audit Collision Loop
    // First, find an existing audit name
    const auditSelector = await page.$('#audit-selector');
    const existingName = await page.evaluate(el => el.options[0].text.split(' (')[0], auditSelector);

    await page.fill('#app-name', existingName);
    await page.click('#create-audit-btn');
    await page.waitForSelector('.modal-overlay');
    await page.click('#modal-confirm'); // "Utwórz nowy"

    await page.waitForSelector('.kawaii-modal');
    await page.screenshot({ path: 'verification/theme_check_v2/collision_dialog.png' });

    // Click "Zmień nazwę" (confirmLabel)
    await page.click('#kawaii-confirm');
    await page.waitForSelector('#kawaii-prompt-input');
    await page.screenshot({ path: 'verification/theme_check_v2/prompt_dialog.png' });

    await browser.close();
})();

File: verification/verify_fixes.py
from playwright.sync_api import Page, expect, sync_playwright
import time

def verify_fixes(page: Page):
    page.goto("http://localhost:3000")

    # Wait for the app to be ready
    page.wait_for_selector("#app-name")

    # 1. Verify Three-State Filters
    # Get the first filter label in the 'Poziom' group
    filter_label = page.locator(".filter-group:has(legend:text('Poziom')) label").first
    filter_input = filter_label.locator("input")

    # Neutral state
    expect(filter_label).not_to_have_class(r"filter-include")
    expect(filter_label).not_to_have_class(r"filter-exclude")

    # Click -> Include
    filter_input.click()
    expect(filter_label).to_have_class(r"filter-include")

    # Click -> Exclude
    filter_input.click()
    expect(filter_label).to_have_class(r"filter-exclude")

    # Click -> Neutral
    filter_input.click()
    expect(filter_label).not_to_have_class(r"filter-include")
    expect(filter_label).not_to_have_class(r"filter-exclude")

    # 2. Verify Save Version Button (Alert for default project)
    def handle_dialog(dialog):
        assert "Najpierw wybierz lub utwórz projekt" in dialog.message
        dialog.dismiss()

    page.on("dialog", handle_dialog)
    page.click("#save-version-btn")

    # 3. Verify App Name sync
    page.fill("#app-name", "Kawaii App Test")
    # Verify it stays there
    expect(page.locator("#app-name")).to_have_value("Kawaii App Test")

    page.screenshot(path="verification/fix_verification_final.png", full_page=True)
    print("Verification successful!")

if __name__ == "__main__":
    with sync_playwright() as p:
        browser = p.chromium.launch(headless=True)
        page = browser.new_page()
        try:
            verify_fixes(page)
        finally:
            browser.close()
